controllers/imagejob/imagejob_controller.go

```diff
@@ -313,11 +338,26 @@ func (r *Reconciler) handleNewJob(ctx context.Context, imageJob *eraserv1alpha1.
 			return err
 		}
 		log.Info("Started "+containerName+" pod on node", "nodeName", nodeName)
+
+		// send pod to podQueue to create thread and wait for PodReady phase
+		podQueue <- pod
```

Comment:
This will not block unless the channel is full, so you could possibly end up with a situation where the `waitGroup.Wait()` call happens before anything has been `Add`ed to the wait group. So you should `waitGroup.Add(1)` just before this line.